packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,236 @@
+	const dateMatches = strToParse.match(datePartRegex);
+
+	if (!dateMatches) {
+		const errMsg = 'DATE - INCORRECT FORMAT';
+		return [undefined, errMsg];
+	}
+
+	const yearString = dateMatches[1];
+	const monthString = dateMatches[2];
+	const dayString = dateMatches[3];
+
+	const day = parseInt(dayString, 10);
+	const month = parseInt(monthString, 10);
+	const monthIndex = month - 1;
+	const year = parseInt(yearString, 10);
+
+	if (month === 0 || month > 12) {
+		const errMsg = 'DATE - INCORRECT MONTH NUMBER';
+		return [undefined, errMsg];
+	}
+
+	if (day === 0) {
+		const errMsg = 'DATE - INCORRECT DAY NUMBER';
+		return [undefined, errMsg];
+	}
+
+	const monthDate = new Date(year, monthIndex);
+	const lastDateOfMonth = lastDayOfMonth(monthDate);
+
+	if (day > getDate(lastDateOfMonth)) {
+		const errMsg = 'DATE - INCORRECT DAY NUMBER RELATIVE TO MONTH';
+		return [undefined, errMsg];
+	}
+
+	const dateValidated = setDate(monthDate, day);
+
+	return [dateValidated];
+}
+
+function extractTime(strToParse) {
+	const timeMatches = strToParse.match(timePartRegex);
+
+	if (!timeMatches) {
+		const errMsg = 'TIME - INCORRECT FORMAT';
+		return [undefined, errMsg];
+	}
+
+	const hoursString = timeMatches[1];
+	const minutesString = timeMatches[2];
+
+	const hours = parseInt(hoursString, 10);
+
+	if (hours >= 24) {
+		const errMsg = 'TIME - INCORRECT HOUR NUMBER';
+		return [undefined, errMsg];
+	}
+
+	const minutes = parseInt(minutesString, 10);
+
+	if (minutes >= 60) {
+		const errMsg = 'TIME - INCORRECT MINUTES NUMBER';
+		return [undefined, errMsg];
+	}
+
+	const timeValidated = hoursAndMinutesToTime(hours, minutes);
+
+	return [timeValidated];
+}
+
+const PROPS_TO_OMIT_FOR_INPUT = ['selectedDateTime', 'onChange', 'onError'];
+
+class InputDateTimePicker extends React.Component {
+	static propTypes = {
+		selectedDateTime: PropTypes.instanceOf(Date),
+		onChange: PropTypes.func,
+		onError: PropTypes.func,
+	};
+
+	constructor(props) {
+		super(props);
+
+		// eslint-disable-next-line
+		console.warn(
+			"UNSTABLE WARNING: The 'InputDateTimePicker' and all the sub components aren't ready to be used in Apps. Code can (will) change outside the release process until it's ready.",
+		);
+
+		this.state = (() => {
+			const selectedDateTime = this.props.selectedDateTime;
+			if (selectedDateTime !== undefined) {
+				const date = startOfDay(selectedDateTime);
+				const hours = getHours(selectedDateTime);
+				const minutes = getMinutes(selectedDateTime);
+				const time = hoursAndMinutesToTime(hours, minutes);
+				const fullDate = startOfMinute(selectedDateTime);
+
+				return {
+					date,
+					time,
+					lastFullDate: fullDate,
+					textInput: getTextDate(date, time),
+				};
+			}
+
+			return {
+				textInput: '',
+			};
+		})();
+
+		this.onChangeInput = this.onChangeInput.bind(this);
+		this.onSubmitPicker = this.onSubmitPicker.bind(this);
+	}
+
+	onSubmitPicker({ date, time }) {
+		this.updateDateTime(date, time);
+	}
```

Comment:
This function is not useful, and introduce more complexity as sometimes we have the date and time as params, and sometimes we have an object containing both.
Remove that and use `updateDateTime`. Change the api of `updateDateTime` to fit what is needed.

Reply:
We already discussed it, there is no complexity at all, even it's the reverse. Each piece is divided from the other, there is no strong coupling between components and between handler/function actions.
- `onSubmitPicker` is the handler, `updateDateTime` is the action.
    => An handler can trigger multiple actions, in fact that's what'll happen
```javascript
onSubmitPicker({ date, time }) {
	this.updateDateTime(date, time);
	this.switchDropdownVisibility(false);
}
```
- The signature of the two are not the same : date and time VS date and time and input string and error message.
- Explicit definition is far more clear about what happens and also what's not happening
    => If I do the following `onSubmit={this.updateDateTime}`, we need to go check in the other component what's the interface of submit, to verify with what args `updateDateTime` will be called.
    => Even worse, if we decide to add an argument to the submit callback, from the other component point of view, this is legitimate, it's an adding, it's not a breaking change, but for this component, it will break because it assumes the interface is immutable.
- The interface of the other component `DateTimePicker` is it's own. It have a coherent input/output interface with an identical object format. Transforming all interface to have exactly the same format value is just a step closer to strong coupling and lead always to : "change one little thing, break everything".
    => In the actual component, date and time variables/args, are always separate, it's totally coherent. And when we use the interface of an another component yes it can be different, but it's not one time like this one time like that, it's always coherent in a component scope point of view (internal method signature (ie not handlers from others components/libs/etc.), always have date and time separate)

In the other hand : the only benefit I see for tying functions together, is to remove some characters. And it's not at all a benefit in itself. It is when it's concise, not when it's obscure.

And I will not have any word about performance for a single function called added, the performance reasons, are always the worst one.

> Programs must be written for people to read, and only incidentally for machines to execute.
> -Abelson & Sussman, Structure and Interpretation of Computer Programs